rslib/src/card_rendering/parser.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::collections::HashMap;
+
+use nom::{
+    branch::alt,
+    bytes::complete::is_not,
+    bytes::complete::tag,
+    character::complete::{anychar, multispace0},
+    combinator::{map, not, recognize, success, value},
+    multi::{many0, many1},
+    sequence::{delimited, pair, preceded, separated_pair, terminated, tuple},
+};
+
+use super::{CardNodes, Node, OtherTag, Tag, TtsTag};
+
+type IResult<'a, O> = nom::IResult<&'a str, O>;
+
+impl<'a> CardNodes<'a> {
+    pub(super) fn parse(mut txt: &'a str) -> Self {
+        let mut nodes = Vec::new();
+        while let Ok((remaining, node)) = node(txt) {
+            txt = remaining;
+            nodes.push(node);
+        }
+
+        Self(nodes)
+    }
+}
+
+impl<'a> Tag<'a> {
+    fn new(name: &'a str, options: Vec<(&'a str, &'a str)>, content: &'a str) -> Self {
+        match name {
+            "tts" => {
+                let mut lang = "";
```

Comment:
Currently, lang can be omitted and the default value is "". Should it be mandatory or should we add a default like "en_US"?

Reply:
I can see arguments for both approaches, but think I'm leaning slightly towards making it required, as that way it's consistent for every language. WDYT?

Reply:
Not sure. Codewise, mandatory options makes things a bit trickier. Currently, `Tag::new()` is infallible.
Maybe we can come back to this after deciding on an error management approach.

Reply:
Sounds good, though the checking doesn't necessarily need to be handled at the parsing stage. With the current code, we require lang already - if the user inputs {{tts :Field}}, they get a popup that shows no voice is available (with a not particularly friendly message)

Reply:
I completely forgot about that, so when I said "mandatory", I was thinking of refusing to parse that tag (into a tts tag) if no language was specified. The pop-up makes much more sense!
So the Rust code is fine here if I understood correctly.